controllers/volsync/vshandler.go

```diff
@@ -578,21 +562,21 @@ func (v *VSHandler) pvcExistsAndInUse(pvcName string, inUsePodMustBeReady bool)
 	return util.IsPVAttachedToNode(v.ctx, v.client, v.log, pvc)
 }
 
-func (v *VSHandler) pvcExists(pvcName string) (bool, error) {
-	_, err := v.getPVC(pvcName)
+func (v *VSHandler) pvcExists(pvcName string) (bool, *corev1.PersistentVolumeClaim, error) {
+	pvc, err := v.getPVC(pvcName)
 	if err != nil {
 		if !kerrors.IsNotFound(err) {
 			v.log.V(1).Info("failed to get PVC")
 
-			return false, err
+			return false, pvc, err
```

Comment:
can be nil, err when getting the pvc failed.

Reply:
We don't care. The precedence is: "error" checked first, then "exists" second, then, lastly "pvc" is used.

Reply:
done